ru/lessons/specifics/plug.md

```diff
@@ -0,0 +1,235 @@
+---
+layout: page
+title: Plug
+category: specifics
+order: 1
+lang: ru
+---
+
+Если вы знакомы с `Ruby`, то можете думать о `Plug` как о комбинации `Rack` и `Sinatra`. Во-первых, `Plug` это набор договорённостей и спецификаций, позволяющих создавать универсальные компонуемые модули, иcпользуемые в веб-приложениях. Во-вторых - адаптеры соединений для различных веб-серверов на платформе `Erlang VM`. Хотя `Plug` и не является частью ядра `Elixir`, это официальный проект `Elixir`.
+
+{% include toc.html %}
+
+## Установка
+
+Plug устанавливается с помощью `mix`. Для установки `Plug` нам необходимо внести два небольших изменения в наш файл `mix.exs`. Первое, что необходимо сделать, это добавить `Plug` и веб-сервер на выбор (мы будем использовать `Cowboy`) в наш файл в качестве зависимостей:
+
+```elixir
+defp deps do
+  [{:cowboy, "~> 1.0.0"},
+   {:plug, "~> 1.0"}]
+end
+```
+
+Во-вторых, нам нужно добавить веб-сервер и `Plug` к нашему `OTP` приложению:
+
+```elixir
+def application do
+  [applications: [:cowboy, :logger, :plug]]
+end
+```
+
+## Спецификация
+
+Чтобы создавать собственные модули `Plug`, нужно придерживаться спецификации. К счастью, нам необходимо реализовать всего две функции: `init/1` и `call/2`.
+
+Функция `init/1` используется для инициализации параметров нашего модуля `Plug`, эти параметры передаются в качестве второго аргумента в функцию `call/2`. В дополнение к инициализированным параметрам, функция `call/2` получает структуру `%Plug.Conn` в качества своего первого аргумента, и ожидается, что она также вернёт соединение (структуру того же типа).
```

Comment:
Why? `Plug.Conn` is the module name but `%Plug.Conn` is the name of the struct defined by this module...